tools/ci/check_grep2.py

```diff
@@ -0,0 +1,337 @@
+
+
+def check_lattice_and_wall_stacking(filename, lines):
+    for line in lines:
+        if re.match(
+            r'"\w+" = \(\n[^)]*?/obj/structure/lattice[/\w]*?,\n[^)]*?/turf/simulated/wall[/\w]*?,\n[^)]*?/area/.+?\)',
+            line,
+        ):
+            return "ERROR: Found a lattice stacked within a wall, please remove them."
+
+
+def check_window_and_wall_stacking(filename, lines):
+    for line in lines:
+        if re.match(
+            r'"\w+" = \(\n[^)]*?/obj/structure/window[/\w]*?,\n[^)]*?/turf/simulated/wall[/\w]*?,\n[^)]*?/area/.+?\)',
+            line,
+        ):
+            return "ERROR: Found a window stacked within a wall, please remove it."
+
+
+def check_airlock_and_wall_stacking(filename, lines):
+    for line in lines:
+        if re.match(
+            r'"\w+" = \(\n[^)]*?/obj/machinery/door/airlock[/\w]*?,\n[^)]*?/turf/simulated/wall[/\w]*?,\n[^)]*?/area/.+?\)',
+            line,
+        ):
+            return "ERROR: Found an airlock stacked within a wall, please remove it."
+
+
+def check_grille_and_window_stacking(filename, lines):
+    for line in lines:
+        if re.match(
+            r'"\w+" = \(\n[^)]*?/obj/structure/grille,\n[^)]*?/obj/structure/window/full[/\w]*?,\n[^)]*?/area/.+\)',
+            line,
+        ):
+            return "ERROR: Found grille above a fulltile window. Please replace it with the proper structure spawner."
+        if re.match(
+            r'"\w+" = \(\n[^)]*?/obj/structure/grille,\n[^)]*?/obj/structure/window/full/plasmareinforced,\n[^)]*?/area/.+\)',
+            line,
+        ):
+            return "ERROR: Found grille above a fulltile plastitanium window. Please replace it with the proper structure spawner."
+
+
+def check_area_varedits(filename, lines):
+    for line in lines:
+        if re.match(r"^/area/.+[{]", line):
+            return "ERROR: Variable editted /area path use detected in a map, please replace with a proper area path."
+
+
+def check_base_turf_type(filename, lines):
+    for line in lines:
+        if re.match(r"/turf\s*[,\){]", line):
+            return "ERROR: Base /turf path use detected in maps, please replace it with a proper turf path."
+
+
+def check_multiple_turfs(filename, lines):
+    for line in lines:
+        if re.match(
+            r'"\w+" = \(\n[^)]*?/turf/[/\w]*?,\n[^)]*?/turf/[/\w]*?,\n[^)]*?/area/.+?\)',
+            line,
+        ):
+            return "ERROR: Multiple turfs detected on the same tile! Please choose only one turf!"
+
+
+def check_multiple_areas(filename, lines):
+    for line in lines:
+        if re.match(r'"\w+" = \(\n[^)]*?/area/.+?,\n[^)]*?/area/.+?\)', line):
+            return "ERROR: Multiple areas detected on the same tile! Please choose only one area!"
+
+
+def check_common_spelling_mistakes(filename, lines):
+    for line in lines:
+        if "nanotransen" in line.lower():
```

Comment:
This feels like it could be made a bit more robust, but it's probably fine for this PR.

Reply:
Ye